src/main/java/com/google/alpollo/AuthenticationServlet.java

```diff
@@ -1,16 +1,21 @@
 package com.google.alpollo;
 
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
-import com.google.appengine.api.users.UserService;
 
 @WebServlet("/auth")
 public class AuthenticationServlet extends HttpServlet {
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         UserService userService = UserServiceFactory.getUserService();
+        String loginUrl = userService.createLoginURL("/");
```

Comment:
From this servlet you return login url, but on the frontend you use it as "Login | Logout" url, it'll be better to return json response with some information about user, if he's logged in and logout url, or logout url if not, so we'll know exactly is the user authenticated or not.